sklearn/meson.build

```diff
@@ -60,6 +60,27 @@ np_dep = declare_dependency(include_directories: inc_np)
 openmp_dep = dependency('OpenMP', language: 'c', required: false)
 
 if not openmp_dep.found()
+  warn_about_missing_openmp = true
+  # On Apple Clang avoid a misleading warning if compiler variables are set.
+  # See https://github.com/scikit-learn/scikit-learn/issues/28710 for more
+  # details. This may be removed if the OpenMP detection on Apple Clang improves,
+  # see https://github.com/mesonbuild/meson/issues/7435#issuecomment-2047585466.
+  if host_machine.system() == 'darwin' and cc.get_id() == 'clang'
+    compiler_env_vars_with_openmp = run_command(py,
+      [
+        '-c',
+        '''
+import os
+
+compiler_env_vars_to_check = ["CPPFLAGS", "CFLAGS", "CXXFLAGS"]
+
+compiler_env_vars_with_openmp = [
+    var for var in compiler_env_vars_to_check if "-fopenmp" in os.getenv(var, "")]
+print(compiler_env_vars_with_openmp)
+'''], check: true).stdout().strip()
+      warn_about_missing_openmp = compiler_env_vars_with_openmp == '[]'
```

Comment:
it would be more _standard_ I guess if the process would return 1 on failure instead of checking program's output. But it doesn't really matter here I guess.

Reply:
Why not but at the same time, the process returning 1 could also be because there is a syntax error in the Python snippet, so you would need additional logic or a different exit code, which would complicate things a bit.

Reply:
although, you could return a non-1 value. But doesn't matter. Merged :)